tests/sources/fixtures/confluence/fixture.py

```diff
@@ -7,26 +7,50 @@
 """
 import io
 import os
-import random
-import string
+import time
 
 from flask import Flask, request
 
-DATA_SIZE = os.environ.get("DATA_SIZE", "small").lower()
-_SIZES = {"small": 1000000, "medium": 2000000, "large": 6000000}
-FILE_SIZE = _SIZES[DATA_SIZE]
-LARGE_DATA = "".join([random.choice(string.ascii_letters) for _ in range(FILE_SIZE)])
+from tests.commons import WeightedFakeProvider
+
+fake_provider = WeightedFakeProvider()
+
+DATA_SIZE = os.environ.get("DATA_SIZE", "medium")
+
+match DATA_SIZE:
+    case "small":
+        SPACE_COUNT = 100
+        SPACE_OBJECT_COUNT = 100
+        ATTACHMENT_COUNT = 3
+    case "medium":
+        SPACE_COUNT = 100
+        SPACE_OBJECT_COUNT = 200
+        ATTACHMENT_COUNT = 5
```

Comment:
These look like a pretty significant change from what the ftest is currently doing. Why not add new `cases` with. these numbers, instead of replacing the current test case?

If I'm reading the code right, before this change, it was effectively:
```
SPACE_COUNT = 4000
SPACE_OBJECT_COUNT = 50
ATTACHMENT_COUNT = 6
```

Reply:
Yes, but the data changed - before it was static size files, now they are different - some are small, some are bigger.

So before sync took 59 seconds and synced 1MB files.

Now sync takes around 78 seconds and syncs files from 100KB to 25MB files.

On the screenshots you see how the memory profiles changed:

Before:

![image](https://github.com/elastic/connectors-python/assets/12238374/2ad021a1-7a9e-4b4d-9f60-f148df8c8823)

After:

![image](https://github.com/elastic/connectors-python/assets/12238374/54f0237c-f7ad-49a0-8087-b8498408454a)

TDLR: if I use the old setup, the sync will take significantly more time - approximately `(4000 * 50 * 6) / (200 * 100 * 5)=12` times more making it ~12-15 minutes.